packages/contracts/contracts/RollupMerkleUtils.sol

```diff
@@ -0,0 +1,268 @@
+        }
+        // Store the new root
+        tree.root = computedNode;
+    }
+
+    /**
+     * @notice Get siblings for a leaf at a particular index of the tree.
+     *         This is used for updates which don't include sibling nodes.
+     * @param _path The path from the leaf to the root / the index of the leaf.
+     * @return The sibling nodes along the way.
+     */
+    function getSiblings(uint _path) public view returns (bytes32[] memory) {
+        bytes32[] memory siblings = new bytes32[](tree.height);
+        bytes32 computedNode = tree.root;
+        for(uint i = tree.height; i > 0; i--) {
+            uint siblingIndex = i-1;
+            (bytes32 leftChild, bytes32 rightChild) = getChildren(computedNode);
+            if (getNthBitFromRight(_path, siblingIndex) == 0) {
+                computedNode = leftChild;
+                siblings[siblingIndex] = rightChild;
+            } else {
+                computedNode = rightChild;
+                siblings[siblingIndex] = leftChild;
+            }
+        }
+        // Now store everything
+        return siblings;
+    }
+
+    /*********************
+     * Utility Functions *
+     ********************/
+    /**
+     * @notice Get our stored tree's root
+     * @return The merkle root of the tree
+     */
+    function getRoot() public view returns(bytes32) {
+        return tree.root;
+    }
+
+    /**
+     * @notice Set the tree root and height of the stored tree
+     * @param _root The merkle root of the tree
+     * @param _height The height of the tree
+     */
+    function setMerkleRootAndHeight(bytes32 _root, uint _height) public {
+        tree.root = _root;
+        tree.height = _height;
+    }
+
+    /**
+     * @notice Store node in the (in-storage) sparse merkle tree
+     * @param _parent The parent node
+     * @param _leftChild The left child of the parent in the tree
+     * @param _rightChild The right child of the parent in the tree
+     */
+    function storeNode(bytes32 _parent, bytes32 _leftChild, bytes32 _rightChild) public {
+        tree.nodes[getLeftSiblingKey(_parent)] = _leftChild;
+        tree.nodes[getRightSiblingKey(_parent)] = _rightChild;
+    }
+
+    /**
+     * @notice Get the parent of two children nodes in the tree
+     * @param _left The left child
+     * @param _right The right child
+     * @return The parent node
+     */
+    function getParent(bytes32 _left, bytes32 _right) internal pure returns(bytes32) {
+        return keccak256(abi.encodePacked(_left, _right));
+    }
+
+    /**
+     * @notice get the n'th bit in a uint.
+     *         For instance, if exampleUint=binary(11), getNth(exampleUint, 0) == 1, getNth(2, 1) == 1
+     * @param _intVal The uint we are extracting a bit out of
+     * @param _index The index of the bit we want to extract
+     * @return The bit (1 or 0) in a uint8
+     */
+    function getNthBitFromRight(uint _intVal, uint _index) public pure returns (uint8) {
```

Comment:
I think this can be `function isRightSibling(...) public pure returns (bool)` because it's not used in any other way.